Libraries/Text/Text/RCTTextView.m

```diff
@@ -63,6 +63,25 @@ - (instancetype)initWithFrame:(CGRect)frame
 }
 
 #if TARGET_OS_OSX // [TODO(macOS ISS#2323203)
+- (void)dealloc
+{
+  [self removeAllTextStorageLayoutManagers];
+}
+
+- (void)removeAllTextStorageLayoutManagers
+{
+  // On macOS AppKit can throw an uncaught exception
+  // (-[NSConcretePointerArray pointerAtIndex:]: attempt to access pointer at index ...)
+  // during the dealloc of NSLayoutManager.  The _textStorage and its
+  // associated NSLayoutManager dealloc later in an autorelease pool.
+  // Manually removing the layout managers from _textStorage prior to release
+  // works around this issue in AppKit.
+  NSArray<NSLayoutManager *> *managers = [_textStorage layoutManagers];
```

Comment:
Can we confirm this makes a copy so we don't crash on mutating a collection while iterating it? If not we should explicitly copy the result of [_textStorage layoutManagers]

Reply:
Its a good call to copy it.  In practice there's only ever a single object in the array but that could change.